scim/resource_service_principal.go

```diff
@@ -83,6 +84,16 @@ func (a ServicePrincipalsAPI) Delete(servicePrincipalID string) error {
 	return a.client.Scim(a.context, "DELETE", servicePrincipalPath, nil, nil)
 }
 
+func (a ServicePrincipalsAPI) DeleteRepos(userID string) error {
```

Comment:
This does not belong do `ServicePrincipalsAPI`, but rather to a resource code. with SDK migration, `ServicePrincipalsAPI` won't exist in this codebase anymore

Reply:
Added path delete code into resource `Delete` callback